### PR TITLE
[Workflows] Add workflowsTemplates server route for template docx

### LINKED ISSUE #2133: [Workflows]Add US_ND form template to backend server
**Is your feature request related to a problem? Please describe.**
We need to request the US_ND early termination form template in the frontend to be able to fill it in and print it using the `docxtemplater` tool.

**Describe the solution you'd like**
Add a state-specific route to the backend that bypasses the auth token and sends the template file. 

**Describe alternatives you've considered**
Considered loading the template as an XML file on the frontend, but I didn't find any tools that would make that a simple process. The tool we're using expects a docx zip file. 

I've done some research into how to load the docx file in via webpack but haven't found anything that reads it in as binary.

**Additional context**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

This is in preparation for the ND early termination form automation. This endpoint will serve the docx template file saved in the `server/assets/workflowsTemplates/US_ND` directory.

I also considered sending this as a buffer but decided against it in case we have different requirements for other forms/templates that might be served from this endpoint. 

The validations check the user state code and filename param. 

NOTE: This is not the final template, I'm still working on the template placeholders here:
https://docs.google.com/document/d/1-SyePvfoWTq2BQR3DIpPyXL6o7vsw5KA/edit

@ohaibbq could you take a look at the template and let me know if I'm missing or misunderstanding any of the sections / placeholders?

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #2133 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -70,6 +73,7 @@ const routesExemptFromJwtValidation = [
   "/_ah/warmup",
   "/_ah/start",
   "/api/demoUser",
+  `${stateApiBaseRoute}workflowsTemplates`,
```
Comment (macfarlandian) [inline server/app.js:76]: `workflows/templates`?

**Review by macfarlandian — COMMENTED** [review]
I did not know this was the solution to our free-text issue and it is truly delightful

**Comment by daschi** [comment]
> I did not know this was the solution to our free-text issue and it is truly delightful

This template is going to be used with the `docxtemplater` tool, you can see a exploration of it on `daschi-docx-exploration` branch!

**Comment by daschi** [comment]
Closing in favor of other PR

**Pull request closed without merging**